include/ginkgo/core/base/array.hpp

```diff
@@ -570,6 +571,31 @@ class Array {
 };
 
 
+/**
+ * Reduce the values in the array
+ *
+ * @tparam The type of the input data
+ *
+ * @param [in] input_arr the input array to be reduced
+ * @param [in] init_val the initial value
+ * @return the reduced value
+ */
+template <typename ValueType>
+ValueType reduce(const Array<ValueType>& input_arr,
+                 const ValueType init_val = 0);
+
+/**
+ * Reduce the values in the array
+ *
+ * @tparam The type of the input data
+ *
+ * @param [in] input_arr the input array to be reduced
+ * @param [out,in] value the reduced value
+ */
+template <typename ValueType>
+void reduce(const Array<ValueType>& input_arr, ValueType* value);
```

Comment:
- Missing description that the initial value of `value` is considered (and that you only consider the first value).
- Why do you have the Input as an `Array`, while the output is only a pointer?